src/cleanup.ts

```diff
@@ -3,48 +3,79 @@ import { Utils } from './utils';
+        return;
+    }
+
+    // Check connection to Artifactory before proceeding with build info post tasks
+    try {
+        core.startGroup('Checking connection to JFrog Artifactory');
+        const pingResult: string = await Utils.runCliAndGetOutput(['rt', 'ping']);
+        if (pingResult !== 'OK') {
+            core.warning('Could not connect to Artifactory. Skipping Build Info post tasks.');
+        }
+    } catch (error) {
+        core.warning(`An error occurred while trying to connect to Artifactory: ${error}. Skipping Build Info post tasks.`);
+        return;
+    } finally {
+        core.endGroup();
+    }
+
     // Auto-publish build info if needed
     try {
-        if (!core.getBooleanInput(Utils.AUTO_BUILD_PUBLISH_DISABLE)) {
+        if (!disableAutoBuildPublish) {
             await collectAndPublishBuildInfoIfNeeded();
+        } else {
+            core.info('Auto build info publish is disabled. Skipping auto build info collection and publishing');
         }
     } catch (error) {
-        core.warning('failed while attempting to publish build info: ' + error);
+        core.warning('failed while attempting to collect and publish build info: ' + error);
     }
+
     // Generate job summary
     try {
-        if (!core.getBooleanInput(Utils.JOB_SUMMARY_DISABLE)) {
+        if (!disableJobSummary) {
             core.startGroup('Generating Job Summary');
             await Utils.runCli(['generate-summary-markdown']);
             await Utils.setMarkdownAsJobSummary();
             core.endGroup();
+        } else {
+            core.info('Job summary is disabled. Skipping job summary generation');
         }
     } catch (error) {
         core.warning('failed while attempting to generate job summary: ' + error);
     }
-    // Cleanup JFrog CLI servers configuration
-    try {
-        core.startGroup('Cleanup JFrog CLI servers configuration');
-        await Utils.removeJFrogServers();
-    } catch (error) {
-        core.setFailed((<any>error).message);
-    } finally {
-        core.endGroup();
-    }
 }
 
 function addCachedJfToPath(): boolean {
-    // Get the JFrog CLI path from step state. saveState/getState are methods to pass data between a step, and it's cleanup function.
-    const jfCliPath: string = core.getState(Utils.JF_CLI_PATH_STATE);
-    if (!jfCliPath) {
-        // This means that the JFrog CLI was not installed in the first place, because there was a failure in the installation step.
-        return false;
+    let version: string = core.getInput(Utils.CLI_VERSION_ARG);
+    if (version == Utils.LATEST_CLI_VERSION) {
+        // If the version is 'latest', we keep it on cache as 100.100.100
+        version = Utils.LATEST_SEMVER;
     }
-    core.addPath(jfCliPath);
-    return true;
+    let jfFileName: string = Utils.getJfExecutableName();
+    let jfrogFileName: string = Utils.getJFrogExecutableName();
+    return Utils.loadFromCache(jfFileName, jfrogFileName, version);
```

Comment:
This method duplicates functionality found in getAndAddCliToPath. Please consider merging the two methods and relocating any general logic from cleanup.ts to utils.ts.